### PR TITLE
[layer] Update Activation layer to LayerV2 

### PR DESCRIPTION
**Pull request by kparichay**
Update activation layer implementation with layerV2 design.
Also add minor updates to input layer.

Enable more unittests for modelfile which depended on activation layers.
Also updated some of the unittests with the change in behavior of the
network graphs.

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: nntrainer/layers/activation_layer.cpp [hunk 1]
```diff
@@ -30,44 +30,64 @@
...
 
-void ActivationLayer::calcDerivative() {
-  Tensor &deriv = net_hidden[0]->getGradientRef();
-  Tensor &ret = net_input[0]->getGradientRef();
-  Tensor &in = net_hidden[0]->getVariableRef();
+void ActivationLayer::calcDerivative(RunLayerContext &context) {
+  Tensor &deriv = context.getIncomingDerivative(SINGLE_INOUT_IDX);
+  Tensor &ret = context.getOutgoingDerivative(SINGLE_INOUT_IDX);
+  Tensor &in = context.getOutput(SINGLE_INOUT_IDX);
+  // Tensor &deriv = net_hidden[0]->getGradientRef();
```
Comment (zhoonit) [inline nntrainer/layers/activation_layer.cpp:49]: Wonder if this is intended comment?

**Review by zhoonit — APPROVED** [review]
LGTM

File: nntrainer/layers/activation_layer.cpp (hunk 1 above)
Comment (kparichay) [reply nntrainer/layers/activation_layer.cpp:49]: my bad, will be removed in rebase.

Comment (kparichay) [reply nntrainer/layers/activation_layer.cpp]: done.

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**